web/client/components/manager/rulesmanager/SimpleSpatialFilter.jsx

```diff
@@ -0,0 +1,209 @@
+                </button>
+            </span>
+        );
+    };
+
+    renderSpatialHeader = () => {
+        const selectedMethod = this.getMethodFromId(this.props.spatialField.method);
+        return (
+            <div className="container-fluid">
+                <Row className="logicHeader filter-field-row filter-field-fixed-row">
+                    <Col xs={6}>
+                        <div><I18N.Message msgId={"queryform.spatialfilter.filterType"}/></div>
+                    </Col>
+                    <Col xs={6}>
+                        <ComboField
+                            itemComponent={(other) =>
+                                (<ComboFieldListItem
+                                    customItemClassName={this.getMethodFromId(other.item) && this.getMethodFromId(other.item).customItemClassName || ""}
+                                    {...other}/>)}
+                            fieldOptions={
+                                this.props.spatialMethodOptions.map((opt) => {
+                                    return LocaleUtils.getMessageById(this.context.messages, opt.name) || opt.name;
+                                })
+                            }
+                            placeholder={LocaleUtils.getMessageById(this.context.messages, "queryform.spatialfilter.combo_placeholder")}
+                            fieldName="method"
+                            fieldRowId={new Date().getTime()}
+                            fieldValue={
+                                LocaleUtils.getMessageById(this.context.messages, selectedMethod ? selectedMethod.name : "") || selectedMethod && selectedMethod.name || ""
+                            }
+                            onUpdateField={this.updateSpatialMethod}/>
+                    </Col>
+                </Row>
+            </div>
+        );
+    };
+    renderSpatialPanel = (drawLabel) => {
+        return (
+            <Panel className="spatial-panel">
+                {this.props.spatialMethodOptions.length > 1 ? this.renderSpatialHeader() : <span/>}
+                { this.props.spatialField.method === "CQL" && (
+                    <RoiCql wkt={this.props.wkt} onChangeFilter={this.cqlChanged}/>
+                )}
+                <Row>
+                    <Col xs={12}>
+                        {drawLabel}
+                    </Col>
+                </Row>
+            </Panel>
+        );
+    };
+    renderButtons = () => {
+        const buttons = [];
+        const showReset = this.props.spatialField.geometry && this.props.spatialField.geometry.coordinates;
+        if (showReset) {
+            buttons.push({
+                glyph: 'clear-filter',
+                tooltipId: "rulesmanager.remove",
+                onClick: () => this.resetSpatialFilter()
+            });
+        }
+        return buttons;
+    };
+    render() {
+        let drawLabel = null;
```

Comment:
(optional) There is no reason to make the renderSpatialPanel parametric. please move this check in renderSpatialPanel (explain the reason in a comment or make the condition a named function or variable that auto-explains the resason of the conditional render